packages/webcomponentsjs/ts_src/platform/event-listener-options.ts

```diff
@@ -0,0 +1,85 @@
+/**
+ * @license
+ * Copyright (c) 2016 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt The complete set of authors may be found
+ * at http://polymer.github.io/AUTHORS.txt The complete set of contributors may
+ * be found at http://polymer.github.io/CONTRIBUTORS.txt Code distributed by
+ * Google as part of the polymer project is also subject to an additional IP
+ * rights grant found at http://polymer.github.io/PATENTS.txt
+ */
+
+export {};
+
+// Older browsers like IE do not support an object as the options parameter
+// to add/removeEventListener.
+// https://connect.microsoft.com/IE/feedback/details/790389/event-defaultprevented-returns-false-after-preventdefault-was-called
+const supportsEventOptions = (() => {
+  let supported = false;
+  const eventOptions = {
+    get capture() {
+      supported = true;
+      return false;
+    },
+  };
+  const listener = () => {};
+  // NOTE: These will be unpatched at this point.
+  window.addEventListener('test', listener, eventOptions);
+  window.removeEventListener('test', listener, eventOptions);
+  return supported;
+})();
+
+const nativeEventTarget = window.EventTarget ?? window.Node;
+
+if (
+  !supportsEventOptions &&
+  'addEventListener' in nativeEventTarget.prototype
+) {
+  const parseEventOptions = (
+    optionsOrCapture?: boolean | AddEventListenerOptions
+  ) => {
+    let capture, once, passive;
+    if (optionsOrCapture && optionsOrCapture instanceof Object) {
+      capture = Boolean(optionsOrCapture.capture);
+      once = Boolean(optionsOrCapture.once);
+      passive = Boolean(optionsOrCapture.passive);
+    } else {
+      capture = Boolean(optionsOrCapture);
+      once = false;
+      passive = false;
+    }
+    return {
+      capture,
+      once,
+      passive,
+    };
+  };
+
+  const origAddEventListener = nativeEventTarget.prototype.addEventListener;
+  const origRemoveEventListener =
+    nativeEventTarget.prototype.removeEventListener;
+
+  nativeEventTarget.prototype.addEventListener = function (
+    type: string,
+    listener: EventListenerOrEventListenerObject | null,
+    options?: boolean | AddEventListenerOptions | undefined
+  ) {
+    const {capture, once} = parseEventOptions(options);
+    const nativeListener = once
+      ? (e: Event) => {
+          origRemoveEventListener.call(this, type, nativeListener, capture);
+          ((listener as EventListenerObject).handleEvent ?? listener)(e);
+        }
+      : listener;
+    origAddEventListener.call(this, type, nativeListener, capture);
+  };
+
+  nativeEventTarget.prototype.removeEventListener = function (
+    type: string,
+    listener: EventListenerOrEventListenerObject | null,
+    options?: boolean | AddEventListenerOptions | undefined
+  ) {
+    const {capture} = parseEventOptions(options);
+    origRemoveEventListener.call(this, type, listener, capture);
```

Comment:
This won't remove the correct listener because `listener` here isn't the same listener that's given to `origAddEventListener` above (`nativeListener`). An important thing to note is that keeping a map from the `listener` parameter to the actual listener given to the native function won't be enough to fix this because of how listeners are deduplicated when they're added based on their options.

The 'formdata' event polyfill has to handle a kind of similar situation and has this [`EventListenerArray` class](https://github.com/webcomponents/polyfills/blob/6f17afd6bf42c7383ade9b9819c378c9f504b735/packages/formdata-event/ts_src/event_listener_array.ts) that tracks the deduplication parts. However, the 'formdata' event polyfill only needs to pay attention to listeners for 'formdata' events and isn't actually wrapping these listeners (it only cares when the number of deduplicated 'formdata' event listeners on a target changes between zero and non-zero) so `EventListenerArray` isn't set up to actually provide a full mapping of user-provided listeners to the 'real' listeners; but it would likely be a good starting point for doing that here. There's also a few things in it that aren't needed here: the `push` and `delete` methods are really the important parts.